### DISCUSSION TITLE
Calling Fit View outside of the Component

### DISCUSSION THREAD

**Discussion by maindotmarcell** [xyflow/xyflow#2532] [Q&A] [answered]
Hi there, I'm looking to obtain fit-view functionality on the initial load of the react flow layout, as well as some events in the app (such as loading different data), i.e. changing the position of the nodes to fit the view within my functions that get triggered based on user interaction (apart from clicking the fit view button). Is there a way to do that?
I want to call the fit view feature inside the callback functions I use when setting new nodes.
Thanks in advance!

**Comment by moklick** [comment, marked as answer]
Hey! You can wrap your app with a [`<ReactFlowProvider>`](https://reactflow.dev/docs/api/react-flow-provider/) and then use `fitView` from [`useReactFlow`](https://reactflow.dev/docs/api/hooks/use-react-flow/). Does it help?

**Reply by maindotmarcell** [reply]
Hi, that's exactly what I needed! Thank you!

**Reply by CSworenI** [reply]
@moklick I have tried this and it's not working for me. I am trying to time it with the opening/closing of an options pane (`aside` element) with the changing of a state (which also controls the viewable space of the grid). I have tried it multiple ways:

With the `onClick` event:
```
onClick={() => [setOptions(false), fitView()]}
```

`useEffect` triggered on the state as described above:
```
	useEffect(() => {
		fitView();
	}, [options]);
```

similar, but per documentation (may be old?):
```
	useEffect(() => {
		window.requestAnimationFrame(() => fitView());
	}, [options]);
```

and another similar, but using the instance of the `ReactFlow` component:
```
	useEffect(() => {
		window.requestAnimationFrame(() => flowInstance.fitView());
	}, [options]);
```

I've also tried each combination without the `window.requestAnimationFrame` clause. None of these seem to work. Can you advise?

**Comment by piyusht1402** [comment]
Hi, I'am unable to use fitview() for my external buttons at all, I have some functions that toggle the layout of the graph but they dont get resized according to the viewport at all. I have tried all the methods for fitview, but most of the time i get an error "fitview is not a function".

**Reply by moklick** [reply]
it's `fitView` with an uppercase "V"

**Reply by piyusht1402** [reply]
Yea I'm sorry for the typo here, I actually meant fitView.

**Comment by FabioRossi-99** [comment]
@moklick 
I tried to use the fitView() in a component inside a SvelteFlowProvider & SvelteFlow but i cannot fit the view.

`
	<SvelteFlowProvider>
		<SvelteFlow
			nodes={finalNodes}
			{edges}
			{nodeTypes}
			{edgeTypes}
			fitView
			zoomOnPinch
			panOnDrag
			maxZoom={2}
			minZoom={0.1}
			bind:fitViewOptions
			connectionLineType={ConnectionLineType.SmoothStep}
			defaultEdgeOptions={{ type: 'smoothstep', animated: true }}
		>
			<Controls 
				position="bottom-right"
				--xy-controls-box-shadow="none"
				--xy-controls-button-background-color="rgb(var(--global-color-background-200))"
				--xy-controls-button-background-color-hover="rgb(var(--global-color-background-300))"
				showLock={false}
				showZoom={false}
			/>
			<SvelteFlowProvider fitView>
				<FitViewNodes bind:needFit bind:fitViewOptions nodes={$finalNodes.map((n) => {return {id: n.id.toString()}})}/>
			</SvelteFlowProvider>
			<Panel position="top-right">
				<div class="backdrop:blur-sm p-3 rounded flex gap-4" style:backdrop-filter="blur(4px)">
					<Icon name="mdi-cog" click on:click={() => (optionDialogOpen = true)} />
					<ReportDownloadDiagramButton bind:nodes={$finalNodes} />
				</div>
			</Panel>
			<Panel position="top-left">
				<slot name="top-left-panel" />
			</Panel>
			<Background
				bind:bgColor={bgColor}
				patternColor="rgb(var(--global-color-contrast-100))"
			/>
		</SvelteFlow>
</SvelteFlowProvider>
` 

I have a dialog with the option to insert a hidden node. When I insert the hidden node the needFit variable change (the needFit is a prop of the component FitViewNodes). In the component FitViewNodes I use the fitView but it seems not working
